projects/pgai/pgai/vectorizer/embeddings.py

```diff
@@ -414,3 +464,85 @@ async def _encode(self, documents: list[str]) -> list[list[int]]:
     @cached_property
     def _encoder(self) -> tiktoken.Encoding:
         return tiktoken.encoding_for_model(self.model)
+
+
+@dataclass
+class OllamaOptions:
+    num_keep: int
+    seed: int
+    num_predict: int
+    top_k: int
+    top_p: float
+    tfs_z: float
+    typical_p: float
+    repeat_last_n: int
+    temperature: float
+    repeat_penalty: float
+    presence_penalty: float
+    frequency_penalty: float
+    mirostat: int
+    mirostat_tau: float
+    mirostat_eta: float
+    penalize_newline: bool
+    stop: Sequence[str]
+
+
+class Ollama(BaseModel, Embedder):
+    """
+    Embedder that uses Ollama to embed documents into vector representations.
+
+    Attributes:
+        implementation (Literal["ollama"]): The literal identifier for this
+            implementation.
+        model (str): The name of the Ollama model used for embeddings.
+    """
+
+    implementation: Literal["ollama"]
+    model: str
+    truncate: bool = True
+    options: OllamaOptions | None = None
+    keep_alive: float | str | None = None
```

Comment:
We also don't expose this parameter in the SQL API yet (`ai.embedding_ollama()`), I'm not sure if we should. The `keep_alive` parameter basically tells ollama how long it should keep the model in memory after an API call. The default is 5 minutes.

Reply:
might be nice to have for performance reasons?